pkg/webui/src/JobList.tsx

```diff
@@ -121,6 +127,9 @@ class JobListImpl extends React.Component<JobListProps, JobListState> {
         const resp = await new Promise<ListJobsResponse>((resolve, reject) => this.props.client.listJobs(req, (err, resp) => !!err ? reject(err) : resolve(resp!)));
         state.jobs = resp.getResultList().map(r => r.toObject());
         state.totalJobs = resp.getTotal();
+        if (newState.search !== this.state.search) {
+            this.startListening();
+        }
```

Comment:
`startListening` uses the state to determine the new search criteria.
We'd need to call `startListening` once the state has been set, e.g.

```TypeScript
const searchChanged = newState.search !== this.state.search;
this.setState(state);

if (searchChanged) {
    this.startListening();
}
```

Reply:
You're right. I'm surprised it worked in my tests anyway!? Gonna change this later ...

I wonder how useful the search comparison is at all? If I understand it correctly it doesn't perform a deep equal, does it? Thus, the only case when it is equal is when both are `undefined`? Right? However, probably better than nothing ...